packages/store-ui/src/molecules/Modal/useTrapFocus.ts

```diff
@@ -0,0 +1,103 @@
+import { useEffect, useRef } from 'react'
+import type { RefObject } from 'react'
+import type { FocusableElement } from 'tabbable'
+import { tabbable } from 'tabbable'
+
+interface TrapFocusParams {
+  sentinelStartRef: RefObject<HTMLElement>
+  trapFocusRef: RefObject<HTMLElement>
+  sentinelEndRef: RefObject<HTMLElement>
+}
+
+/**
+ * Element that will maintain the focus inside, focus the first element,
+ * and focus back the element that was focus when useTrapFocus is triggered.
+ *
+ * Inspired by Material UI TrapFocus https://github.com/mui-org/material-ui/blob/master/packages/mui-core/src/Unstable_TrapFocus/Unstable_TrapFocus.js
+ */
+const useTrapFocus = ({
+  trapFocusRef,
+  sentinelStartRef,
+  sentinelEndRef,
+}: TrapFocusParams) => {
+  const tabbableNodesRef = useRef<FocusableElement[]>()
+  const lastKeyDownRef = useRef<KeyboardEvent>()
+
+  const nodeToRestoreRef = useRef<HTMLElement | null>(
+    document.hasFocus() ? (document.activeElement as HTMLElement) : null
+  )
+
+  // Focus back on the element that was focused before the trapFocusNode
```

Comment:
What is trapFocusNode?

Reply:
I want to say this: "focus back the element that was focused when useTrapFocus is triggered."